compiler/vm/vm.nim

```diff
@@ -1156,86 +1448,182 @@ proc rawExecute(c: PCtx, start: int, tos: PStackFrame): TFullReg =
         Skipping over generic insts results in some generics failing
         Realistically we need a tyGenericAlias instead of using tyGenericInst then reasoning into it
       ]#
-      regs[ra].intVal = ord(regs[rb].node.typ.sameTypeOrNil(regs[rc].node.typ, {ExactTypeDescValues, ExactGenericParams}))
+      doAssert false
+      #regs[ra].intVal = ord(regs[rb].node.typ.sameTypeOrNil(regs[rc].node.typ, {ExactTypeDescValues, ExactGenericParams}))
       # The types should exactly match which is why we pass `{ExactTypeDescValues, ExactGenericParams}`.
       # This solves checks with generics.
     of opcXor:
-      decodeBC(rkInt)
+      decodeBC(akInt)
       regs[ra].intVal = ord(regs[rb].intVal != regs[rc].intVal)
     of opcNot:
-      decodeB(rkInt)
-      assert regs[rb].kind == rkInt
+      decodeB(akInt)
+      # TODO: proper error reporting?
+      #assert regs[rb].kind == rkInt
       regs[ra].intVal = 1 - regs[rb].intVal
     of opcUnaryMinusInt:
-      decodeB(rkInt)
-      assert regs[rb].kind == rkInt
+      decodeB(akInt)
+      
+      # TODO: proper error reporting?
+      #assert regs[rb].kind == rkInt
       let val = regs[rb].intVal
       if val != int64.low:
         regs[ra].intVal = -val
       else:
         stackTrace(c, tos, pc, reportSem(rsemVmOverOrUnderflow))
     of opcUnaryMinusFloat:
-      decodeB(rkFloat)
-      assert regs[rb].kind == rkFloat
+      decodeB(akFloat)
+      
+      # TODO: proper error reporting?
+      #assert regs[rb].kind == rkFloat
       regs[ra].floatVal = -regs[rb].floatVal
     of opcBitnotInt:
-      decodeB(rkInt)
-      assert regs[rb].kind == rkInt
+      decodeB(akInt)
+      
+      # TODO: proper error reporting?
+      #assert regs[rb].kind == rkInt
       regs[ra].intVal = not regs[rb].intVal
     of opcEqStr:
-      decodeBC(rkInt)
-      regs[ra].intVal = ord(regs[rb].node.strVal == regs[rc].node.strVal)
+      decodeBC(akInt)
+      regs[ra].intVal = ord(regs[rb].atomVal.strVal == regs[rc].atomVal.strVal)
     of opcLeStr:
-      decodeBC(rkInt)
-      regs[ra].intVal = ord(regs[rb].node.strVal <= regs[rc].node.strVal)
+      decodeBC(akInt)
+      regs[ra].intVal = ord(regs[rb].atomVal.strVal <= regs[rc].atomVal.strVal)
     of opcLtStr:
-      decodeBC(rkInt)
-      regs[ra].intVal = ord(regs[rb].node.strVal < regs[rc].node.strVal)
+      decodeBC(akInt)
+      regs[ra].intVal = ord(regs[rb].atomVal.strVal < regs[rc].atomVal.strVal)
     of opcLeSet:
-      decodeBC(rkInt)
-      regs[ra].intVal = ord(containsSets(c.config, regs[rb].node, regs[rc].node))
+      # a = b is subset of c
+      decodeBC(akInt)
+
+      var ret = true
+
+      # TODO: this is slow. Improve
```

Comment:
I think the fundamental fix is a combination of vmgen and set data encoding

Reply:
The support for sets is just something I hacked together as quickly as possible so I have something working